### ISSUE TITLE
A player keeps his class outside the arena

### ISSUE THREAD

**Issue by Syfers96** [alkarinv/BattleArena#168]
A players are guard equipment outside an arena when he leaves returning to the menu when it dies or is it based on title screen instead of respawn and it reconnects with equipment. Yet I have been clean inventory dying and leaving it. Show toi fix it ?

**Comment by alkarinv** [comment]
Try using the most recent.  http://dev.bukkit.org/media/files/701/665/BattleArena.jar

I did a bunch of changes,if that doesn't fix let me know and I'll work with you to fix.

**Comment by ghost** [comment]
I have tried BattleArena v3.8.8 and cause many times woolTeams option is not work.

Moreover, BattleArena v3.8.8 doesn't fix if player died in battleground and back to title screen then reconnect who will get his equipment back and stay in the battleground. It's possible to take equipment out of battleground by collect drops items and wait for battleground complete.

**Comment by alkarinv** [comment]
Alright, what is your IP? If you don't mind I'd like to use you as a way to fix this inventory problem.  I'm sure if I can be on a server with the problem I can fix.

woolTeams should definitely work why dont you paste me the config of the game you are trying to do wool teams with.

**Comment by ghost** [comment]
Battleground:
    enabled: true
    joinType: Queue
    database: bg
    rated: true
    prefix: "&c[Battleground]&e"
    victoryCondition: highestKills
    matchTime: 240
    nTeams: 2
    teamSize: 3-5
    defaults:
        options: [woolTeams]
    preReqs:
        options: [clearInventory]
    onEnter:
        options: [storeAll]
    onLeave:
        options: [restoreAll]
    onJoin:
        options: [pvpOff]
    onPrestart:
        options: [pvpOff, teleportIn, dropItemOff]
    onStart:
        options: [pvpOn, dropItemOff]
    onComplete:
        options: [teleportOut ]
    onSpawn:
        options: [health=20, hunger=20]
        giveClass:
            default: assault
    onDeath:
        options: [respawn, clearInventory]
    winner:
        options: [enchants, health=20, hunger=20]
        enchants:
            - speed:1:600
            - resistance:1:600

This config "woolTeams" works on v3.8.7.5 but doesn't work perfect on v3.8.8

My owner server timezone is UTC+8 for now are very few player online but battleground need 6 player to start. So I create a test server with same battleground config for you but just four player can start.

To test this bug four player is required because two player game will complete when one the other died.

Test Server IP: 114.44.12.100

**Comment by ghost** [comment]
BattleArena v3.8.8 full tested result:
4 player in battleground by previously post config.
1. At first, everyone doesn't wool head.
   In v3.8.7.5 they will be wool head and put their helmet in inventory if they have.
2. One player has been slain then back to title screen and reconnected. That player will auto respawn with his owned item and stay in the place where are he died. 
   In v3.8.7.5, when the battleground complete by time limit that player will be teleported to previously location.
   In v3.8.8, when the battleground complete by time limit that player still stay in the battleground.
3. One player has been slain then respawn. That player will spawned in the battleground spawn point with wool head but no equipment.
   In v3.8.7.5, player will spawned with wool head and default equipment.

**Comment by alkarinv** [comment]
So you are 1.5.1 I see. I'm trying to connect but I need to find some way to downgrade first.

I consider any inventory issues really serious, so please hit me up on irc, (links are on my bukkit pages).

**Comment by ghost** [comment]
I have update CraftBukkit  to 1.5.2 R0.1 and I also discover the battlearena plugin is auto update to v3.8.8.1.1
But unfortunately, the problem is same with my previously post.
Player still can steal item by method 2.

I also tested Default config on battleground and problem is same.

For now v3.8.7.5 is better version for wool head but must set dropItemOff to prevent steal item from battleground.
Also need to warning player for they item probable lose through click "back to title screen" and slain by fool.

**Comment by alkarinv** [comment]
Yeah, 1.5.2 would not fix your problem. I have an important paper due tomorrow.. But can we set up a time to fix all of these problems this weekend?  I'm GMT -5:00.

**Comment by ghost** [comment]
Sure, I'd love to help you.

**Comment by ghost** [comment]
I have free time from 5/18 06:00. (GMT -5:00)

**Comment by alkarinv** [comment]
I'll be trying occasionally to hop on, also if you can come onto IRC, just say the name "alkarin" and it will pop up a message for me.

**Comment by alkarinv** [comment]
Try this,

http://dev.bukkit.org/media/files/703/452/BattleArena.jar

I think that will fix your crash and stay in the arena bug.

**Comment by ghost** [comment]
I have tried BattleArene v3.8.8.1.5 and the title screen bug is still exist.
Wool head is more strange.
Wool head test result:

In v3.8.7.5 they will be wool head and put their helmet in inventory if they have.
In v3.8.8.1.1 they won't wool head.
In v3.8.8.1.5 they won't wool head but they will get one wool in their inventory and can't change class cause changed item.

So, I want to fix wool head by remove helmet in class.yml.
After setting, they neither wear wool head nor wear helmet but they can change class.

If player respawn in arena then this two situation will be automatically fixed.

In v3.8.8.1.5 player respawn will get default class which better than v3.8.8.1.1.
